vm/http.go

```diff
@@ -54,23 +57,64 @@ func builtinHTTPClassMethods() []*BuiltInMethodObject {
 			Name: "get",
 			Fn: func(receiver Object) builtinMethodBody {
 				return func(t *thread, args []Object, blockFrame *callFrame) Object {
-					var path string
 
-					domain := args[0].(*StringObject).value
+					uri, err := url.Parse(args[0].(*StringObject).value)
 
 					if len(args) > 1 {
-						path = args[1].(*StringObject).value
+						var arr []string
+
+						for _, v := range args[1:] {
+							arr = append(arr, v.(*StringObject).value)
+						}
+
+						uri.Path = path.Join(arr...)
+					}
+
+					resp, err := http.Get(uri.String())
+
+					if err != nil {
+						return t.vm.initErrorObject(InternalError, err.Error())
+					}
+					if resp.StatusCode != http.StatusOK {
+						return t.vm.initErrorObject(InternalError, resp.Status)
+					}
+
+					content, err := ioutil.ReadAll(resp.Body)
+					resp.Body.Close()
+
+					if err != nil {
+						return t.vm.initErrorObject(InternalError, err.Error())
 					}
 
-					if !strings.HasPrefix(path, "/") {
-						path = "/" + path
+					return t.vm.initStringObject(string(content))
+				}
+			},
+		}, {
+			// Sends a POST request to the target with type header and body. Returns the HTTP response as a string.
+			Name: "post",
+			Fn: func(receiver Object) builtinMethodBody {
+				return func(t *thread, args []Object, blockFrame *callFrame) Object {
+					if len(args) != 3 {
+						return t.vm.initErrorObject(ArgumentError, "Expect 3 arguments. got=%v", strconv.Itoa(len(args)))
 					}
 
-					resp, err := http.Get(domain + path)
+					uri, err := url.Parse(args[0].(*StringObject).value)
+					if err != nil {
+						return t.vm.initErrorObject(ArgumentError, err.Error())
+					}
+
+					contentType := args[1].(*StringObject).value
+
+					body := args[2].(*StringObject).value
+
+					resp, err := http.Post(uri.String(), contentType, strings.NewReader(body))
 
 					if err != nil {
 						return t.vm.initErrorObject(InternalError, err.Error())
 					}
+					if resp.StatusCode != http.StatusOK {
+						return t.vm.initErrorObject(InternalError, resp.Status)
```

Comment:
BTW This error can be more specific, like `HTTPError`. You can create new error type in `error.go`. But let's merge this first.